webui/src/components/RepositoryListPage.js

```diff
@@ -49,29 +70,37 @@ const RepositoryList = ({ list, paginate }) => {
     }
 
     return (
-        <div>
-            {list.payload.results.map(repo => (
-                <Row key={repo.id}>
-                    <Col className={"mb-2 mt-2"}>
-                        <Card>
-                            <Card.Body>
-                                <h5><Link to={`/repositories/${repo.id}/tree`}>{repo.id}</Link></h5>
-                                <p>
-                                    <small>
-                                        created at <code>{moment.unix(repo.creation_date).toISOString()}</code> ({moment.unix(repo.creation_date).fromNow()})<br/>
-                                        default branch: <code>{repo.default_branch}</code>,{' '}
-                                        storage namesapce: <code>{repo.storage_namespace}</code>
-                                    </small>
-                                </p>
-                            </Card.Body>
-                        </Card>
-                    </Col>
-                </Row>
-            ))}
-            {paginator}
-        </div>
+        <>
+            <div>
+                {list.payload.results.map(repo => (
+                    <Row key={repo.id}>
+                        <Col className={"mb-2 mt-2"}>
+                            <Card>
+                                <Card.Body>
+                                <Button className="float-right" onClick={() => {
+                                                    setSelectedRepo(repo.id);
+                                                    setShowDeleteModal(true);
+                                                    }} 
+                                        variant="danger"><TrashIcon/> Delete Repository</Button>
+                                    <h5><Link to={`/repositories/${repo.id}/tree`}>{repo.id}</Link></h5>
+                                    <p>
+                                        <small>
+                                            created at <code>{moment.unix(repo.creation_date).toISOString()}</code> ({moment.unix(repo.creation_date).fromNow()})<br/>
+                                            default branch: <code>{repo.default_branch}</code>,{' '}
+                                            storage namesapce: <code>{repo.storage_namespace}</code>
+                                        </small>
+                                    </p>
+                                </Card.Body>
+                            </Card>
+                        </Col>
+                    </Row>
+                ))}
+                {paginator}
+            </div>
+            <ConfirmationModal show={showingDeleteModal} onHide={() => {setShowDeleteModal(false)}} msg={`are you sure you wish to delete repository ${selectedRepo}?`} onConfirm={deleteRepo}/>
```

Comment:
Since deleting a repository is probably the most destructive behavior we have in the system (and the one hardest to recover from), perhaps noting *what actually gets deleted* is required here. Example text:

```
Are you sure you wish to delete repository ${selectedRepo}?
This action cannot be undone. This will delete the following:

- All commits
- All branches
- All tags
- All repository configuration

Data in the underlying object store will not be deleted by this action. 

Please type ${selectedRepo} to confirm:
+------------------------------------------+
|                                          |
+------------------------------------------+

```

Reply:
Screenshot of what deleteing a repo on GitHub looks like for further reference:

![image](https://user-images.githubusercontent.com/205955/107140540-62fa1980-692b-11eb-813a-2bdf0b860f3a.png)